pkg/inject/uninject.go

```diff
@@ -59,23 +59,26 @@ func (conf *ResourceConfig) uninjectPodSpec(report *Report) {
 }
 
 func uninjectObjectMeta(t *metav1.ObjectMeta, report *Report) {
-	newAnnotations := make(map[string]string)
-	for key, val := range t.Annotations {
-		if !strings.HasPrefix(key, k8s.Prefix) ||
-			(key == k8s.ProxyInjectAnnotation && val == k8s.ProxyInjectDisabled) {
-			newAnnotations[key] = val
-		} else {
-			report.Uninjected.Proxy = true
-		}
+	//do not uninject meta if this is a control plane component
```

Comment:
This is pretty confusing so I think it warrants a more extensive comment explaining the context.  Control plane components must only ever be manually injected, so they will never have the `linkerd.io/inject` annotation.  Furthermore, since these components are part of linkerd itself, we don't want to strip off the other `linkerd.io/*` annotations such as `linkerd.io/created-by` when uninjecting.

Although.... maybe we DO want to strip off `linkerd.io/proxy-version`?  wdyt?

Reply:
Yes, I will amend that, good call. Wrt to the `linkerd.io/proxy-version` my view is that there is no need to remove it. I assume if the value of the annotation has changed before the nject, then it will be updated. Isn't that correct? 

Also, just to confirm my understanding, from end user perspective, `uninject` as a command would probably never be used on control plane components by a user, correct. Its only used as part of `inject`, which in turn is used not to really inject anything, but to update the injected container such as in the case of adding a debug sidecar. I wonder whether I am the only one who finds all that a bit confusing.

Reply:
>  I wonder whether I am the only one who finds all that a bit confusing.

I'm pretty sure everyone finds this all very confusing.